falcon/routing/static.py

```diff
@@ -15,37 +15,48 @@ def _open_range(file_path, req_range):
         file_path (str): Path to the file to open.
         req_range (Optional[Tuple[int, int]]): Request.range value.
     Returns:
-        tuple: Two-member tuple of (stream, content-range). If req_range is
-            ``None`` or ignored, content-range will be ``None``; otherwise,
-            the stream will be appropriately seeked and possibly bounded,
-            and the content-range will be a tuple of (start, end, size).
+        tuple: Three-member tuple of (stream, content-length, content-range).
+            If req_range is ``None`` or ignored, content-range will be
+            ``None``; otherwise, the stream will be appropriately seeked and
+            possibly bounded, and the content-range will be a tuple of
+            (start, end, size).
     """
     fh = io.open(file_path, 'rb')
+    size = os.fstat(fh.fileno()).st_size
     if req_range is None:
-        return fh, None
+        return fh, size, None
 
     start, end = req_range
-    size = os.fstat(fh.fileno()).st_size
     if size == 0:
-        # Ignore Range headers for zero-byte files; just serve the empty body
-        # since Content-Range can't be used to express a zero-byte body
-        return fh, None
+        # NOTE(tipabu): Ignore Range headers for zero-byte files; just serve
```

Comment:
For reference: @tipabu I prefixed your comments with your GH nick, and refactored the code slightly to work around issues with buggy `sendfile` implementations in many popular WSGI servers such as Gunicorn, uWSGI, etc.